src/main/java/org/sagebionetworks/bridge/validators/DemographicValuesValidator.java

```diff
@@ -0,0 +1,173 @@
+package org.sagebionetworks.bridge.validators;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang3.LocaleUtils;
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+import org.sagebionetworks.bridge.models.studies.Demographic;
+import org.sagebionetworks.bridge.models.studies.DemographicValue;
+import org.sagebionetworks.bridge.models.studies.DemographicValuesValidationConfiguration;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * This is NOT a validator to check if demographics have malformed inputs. For
+ * those validators, see DemographicUserValidator, DemographicValidator, and
+ * DemographicValueValidator.
+ * 
+ * This validator validates the values in demographics to ensure they meet
+ * user-specified restrictions listed in a
+ * DemographicValuesValidationConfiguration which is stored in an app config
+ * element.
+ * 
+ * This does not use the Spring validator framework because errors are stored in
+ * the validated object (Demographic) directly instead of being returned.
+ */
+public class DemographicValuesValidator {
+    private static final String DEMOGRAPHICS_ENUM_DEFAULT_LANGUAGE = "en";
+    private static final String INVALID_DATA = "invalid data";
+    // currently not exposing configuration errors
+    public static final String INVALID_CONFIGURATION = INVALID_DATA;
+    private static final String INVALID_CONFIGURATION_BAD_LANGUAGE_CODE = INVALID_CONFIGURATION;
+    private static final String INVALID_CONFIGURATION_MIN_LARGER_THAN_MAX = INVALID_CONFIGURATION;
```

Comment:
Mentioned in code review that we shouldn't hide invalid configuration. If we're moving this to its own API, we can use the same Validator pattern we use for other APIs and ensure that our configuration as always valid.